src/screens/register/RegisterScreen.tsx

```diff
@@ -0,0 +1,186 @@
+            ) : (
+              <Entypo name='eye' size={24} color={COLORS.NIGHTLIGHT_GRAY} />
+            )}
+          </Pressable>
+        </View>
+
+        <View style={RegisterScreenStyles.passwordContainer}>
+          <Controller
+            control={control}
+            render={({ field: { onChange, onBlur, value } }) => (
+              <TextInput
+                style={RegisterScreenStyles.passwordInput}
+                placeholder='Confirm password'
+                secureTextEntry={isConfirmPasswordHidden}
+                onBlur={onBlur}
+                onChangeText={value => onChange(value)}
+                value={value}
+              />
+            )}
+            name='confirmPassword'
+            rules={{ required: true }}
+          />
+          <Pressable
+            style={RegisterScreenStyles.viewPasswordButton}
+            onPress={() =>
+              setIsConfirmPasswordHidden(!isConfirmPasswordHidden)
+            }>
+            {isConfirmPasswordHidden ? (
+              <Entypo
+                name='eye-with-line'
+                size={24}
+                color={COLORS.NIGHTLIGHT_GRAY}
+              />
+            ) : (
+              <Entypo name='eye' size={24} color={COLORS.NIGHTLIGHT_GRAY} />
+            )}
+          </Pressable>
+        </View>
+        <Controller
+          control={control}
+          render={({ field: { onChange, onBlur, value } }) => (
+            <TextInput
+              style={RegisterScreenStyles.phoneInput}
+              placeholder='Phone'
+              keyboardType='phone-pad'
+              secureTextEntry={isConfirmPasswordHidden}
+              onBlur={onBlur}
+              onChangeText={value => onChange(value)}
+              value={value}
+            />
+          )}
+          name='phone'
+          rules={{ required: true }}
+        />
+        <Pressable style={RegisterScreenStyles.signInButton}>
+          <Text
+            style={RegisterScreenStyles.signInButtonText}
+            onPress={handleSubmit(onSubmit)}>
+            Sign Up
+          </Text>
+        </Pressable>
+        {/* End form */}
+        <Text style={RegisterScreenStyles.continueWithText}>
+          Or continue with
+        </Text>
+        <Pressable style={RegisterScreenStyles.googleSignInButton}>
+          <Image
+            source={require('../../../assets/googleIcon.png')}
+            style={RegisterScreenStyles.googleIcon}
+          />
+          <Text style={RegisterScreenStyles.googleSignInButtonText}>
+            Sign up with Google
+          </Text>
+        </Pressable>
+        <Text style={RegisterScreenStyles.hasAccountText}>
+          Have an account?&nbsp;&nbsp;
+          <Text
+            onPress={() => setIsLogin(true)}
+            style={RegisterScreenStyles.hasAccountLink}>
+            Login now
+          </Text>
+        </Text>
+      </SafeAreaView>
+    </TouchableWithoutFeedback>
+  );
+};
+
+export default RegisterScreen;
```

Comment:
there seems to be a lot of reused code from LoginScreen.tsx......................refactor to use conditional rendering in one screen :)